docs/develop/best-practices/react.md

```diff
@@ -16,13 +41,176 @@ class MyAccordion extends Component {}
   }
 ```
 
+### On Overridable
+
+See the [How to override UI React components](../howtos/override_components.md) how-to for detailed explanations on the library itself.
+
+The library allows developers to mark components that can be overridden. While it is very easy to use, it can also lead to subtle errors.
+
+#### Overridable components naming
+
+The `id` of each Overridable component should use the following pattern:
+
+```javascript
+<Overridable id="[InvenioModuleName].[ReactApp].[Component].[Element]" name={name}>
+```
+
+- `[InvenioModuleName]`: the Invenio module name, e.g. `InvenioCommunities` or `InvenioAppRDM`.
+- `[ReactApp]`: the name of the web page where the ReactApp will be rendered, e.g. `MembersList`, `MyDashboard`, `MyUploads`, `MyCommunities`, `DepositForm`. There is no exact naming here, try to be consistent with the existing ones.
+- `[Component]`: the name of the component in the React app, e.g. `ResultsList`, `SortBy`, `ListItem`, `BtnAccept`.
+- `[Element]`: the name of the UI section inside the component. `Layout` is normally used of the entire Overridable component, `Title` might be used in an inner section, in case of large components:
+
+  ```javascript
+  <Overridable id="InvenioCommunities.InvitationsList.ListItem.Layout">
+    ...
+      <Overridable id="InvenioCommunities.InvitationsList.ListItem.Title">
+  ```
+
+Names should be CamelCase.
+
+#### Clear interfaces
+
+Make sure that your component APIs are very clear. Why? Because when re-implementing a component, the developer should clearly understand what `props` are passed to the Overridable component.
+
 ❌ DON'T
 
 ```javascript
-import React, {Component} from 'react';
+import React, { Component } from "react";
 
-class MyAccordion extends Component {}
-  render(){
-    <div className="ui accordion" key={agg.title}></div>
-  }
+export class ErrorPage extends Component {
+  render() {
+    const { errorCode, errorMessage, error, children } = this.props;
+    return (
+      <Overridable id="..." {...this.props}>
```

Comment:
One case we need to clarify and follow a common strategy is about the prop drilling of SUI underlying components. In a lot of components, we have an open `ui` prop that we are passing as is down to SUI components. Is this something that we are fine with leave it flexible or should we make it stricter? e.g maybe we can import the prop type from SUI([example](https://github.com/inveniosoftware/react-invenio-deposit/blob/af69a468aa200270a80348ce08a83af0b674ef9c/src/lib/components/PreviewButton.js#L32))

Reply:
done!